### PR TITLE
Initial content

### PR DESCRIPTION
**Pull request by ferricoxide**
This PR includes logic for:

- Ensuring the Cribl-Edge software's dependencies installed
- Ensuring the Cribl-Edge software gets exctracted to the desired directory
- Sets the Cribl-Edge Run-mode
- Ensuring the Cribl-Edge software can log into /var/log
- Ensuring the Cribl-Edge software _does_ log into /var/log
- Sets the Cribl-Edge boot-mode to systemd
- Starts the Cribl-Edge systemd Unit

Presumably, will need to further add:

- `fapolicyd` logic (to support hardened-EL8 installations)
- `firewalld` exemptions (to allow remote management-node to contact the service-listener)
- Client-configuration so that the `cribl-edge` service knows how to reach out to its management node

The `firewalld` and "client-configuration" _may_ be mutually-exclusive. Still digging through documents to see what the node/manager relationship-requirements are. Will need information about the development environment's management-server to configure either or both.

### CODE REVIEW

File: cribl-edge/config/log.sls [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    - require:
+      - file: Cribl-Edge OS Log-Dir Setup
+
+Cribl-Edge OS Log-Dir Setup:
+  file.directory:
+    - group: 'root'
+    - mode: '0700'
+    - name: '/var/log/cribl-edge'
+    - require:
+      - archive: 'Cribl-Edge Archive Extracted'
```
Comment (lorengordon) [inline cribl-edge/config/log.sls:24]: Will run into a bit of an ordering issue, if this state is called directly, e.g. `salt-call state.sls cribl-edge.config.log`. The way to work around that problem is to include the sls that defines this "archive" state. For an example, see this setup in the forescout formula:

https://github.com/plus3it/forescout-secure-connector-formula/blob/35705d6f49464956a79d67455c797c39d03a999f/forescout-secure-connector/service/running.sls#L3-L8

File: cribl-edge/service/running.sls [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      - '[[ $( systemctl is-enabled cribl-edge.service ) == "enabled" ]]'
+
+Cribl-Edge Start systemd Unit:
+  service.running:
+    - enable: True
+    - name: 'Cribl-Edge.service'
+    - require:
+      - cmd: 'Cribl-Edge Configure systemd Unit'
+    - unless:
+      - '[[ $( systemctl is-active cribl-edge.service ) == "active" ]]'
```
Comment (lorengordon) [inline cribl-edge/service/running.sls]: This _shouldn't_ be need. It is expected that `service.running` is already stateful.

**Review by lorengordon — COMMENTED** [review]
I don't see a defaults.yaml? That might at least be valuable for the `package.install_path` option?

It would also be useful to add `clean` sls states, to make it easy to start from scratch.

**Comment by ferricoxide** [comment]
> I don't see a defaults.yaml? That might at least be valuable for the `package.install_path` option?

Added and tested (the `source`/`source_hash` now point to the Cribl CDN)

> It would also be useful to add `clean` sls states, to make it easy to start from scratch.

Working that, next.

File: cribl-edge/config/log.sls (hunk 1 above)
Comment (ferricoxide) [reply cribl-edge/config/log.sls:24]: Ok, I _think_ we're good on this. I ran:

1. `salt-call -c /opt/watchmaker/salt/ state.apply cribl-edge`
1. `salt-call -c /opt/watchmaker/salt/ state.apply cribl-edge.clean`
1. `salt-call -c /opt/watchmaker/salt/ state.apply cribl-edge.config.log`

With no errors.

File: pillar.exampl.yml
Comment (lorengordon) [inline pillar.exampl.yml:1]: Filename should be `pillar.example.yml`....

File: cribl-edge/config/log.sls [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    - onlyif:
+      - '[[ -d {{ cribl_edge.package.install_path }}/log ]]'
+    - require:
+      - file: Cribl-Edge OS Log-Dir Setup
+
+Cribl-Edge OS Log-Dir Setup:
+  file.directory:
+    - group: 'root'
+    - mode: '0700'
+    - name: '/var/log/cribl-edge'
```
Comment (lorengordon) [inline cribl-edge/config/log.sls]:
```suggestion
    - name: '{{ cribl_edge.package.real_log_dir }}'
```

File: cribl-edge/config/log.sls [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Cribl-Edge Symlink to OS Log-Dir:
+  file.symlink:
+    - group: 'root'
+    - makedirs: True
+    - mode: '0755'
+    - name: '{{ cribl_edge.package.install_path }}/log'
+    - require:
+      - file: 'Cribl-Edge Nuke Default Log-Dir'
+    - target: '/var/log/cribl-edge'
```
Comment (lorengordon) [inline cribl-edge/config/log.sls]:
```suggestion
    - target: '{{ cribl_edge.package.real_log_dir }}'
```

File: pillar.exampl.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+---
+cribl-edge:
+  lookup:
+    package:
+      archive:
+        source: https://FQDN/<PATH_TO_CRIBL-EDGE_ARCHIVE_FILE>.tgz
+        source_hash: source_hash_url_or_value
+      install_path: <FILESYSTEM_INSTALL_LOCATION>
```
Comment (lorengordon) [inline pillar.exampl.yml]: Update with new options:

```suggestion
    package:
      archive:
        source: https://FQDN/<PATH_TO_CRIBL-EDGE_ARCHIVE_FILE>.tgz
        source_hash: source_hash_url_or_value
      install_path: <FILESYSTEM_INSTALL_LOCATION>
      real_log_dir: /var/log/cribl-edge

    service:
      name: cribl-edge.service
      unit_file: /etc/systemd/system/cribl-edge.service
```

File: cribl-edge/package/clean.sls [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{#- Get the `tplroot` from `tpldir` #}
+{%- set tplroot = tpldir.split('/')[0] %}
+
+{%- from tplroot ~ "/map.jinja" import mapdata as cribl_edge with context %}
+
+Cribl-Edge App-Dir Removed:
+  file.absent:
+    - name: {{cribl_edge.package.install_path }}
+    - onlyif:
+      - '[[ -d {{cribl_edge.package.install_path }} ]]'
```
Comment (lorengordon) [inline cribl-edge/package/clean.sls]: I don't think the onlyif is needed for the clean right? That check was only needed when creating the logdir setup, to stop it from deleting the symlink. For this, the statefulness of `file.absent` should be enough?

```suggestion
```

**Review by lorengordon — COMMENTED** [review]
Looks pretty clean I think. The new setup is nice. Just a few comments to put a bow on it, and then ready to go

File: pillar.exampl.yml
Comment (ferricoxide) [reply pillar.exampl.yml:1]: Fixed

File: cribl-edge/config/log.sls (hunk 3 above)
Comment (ferricoxide) [reply cribl-edge/config/log.sls]: Fixed.

File: pillar.exampl.yml (hunk 5 above)
Comment (ferricoxide) [reply pillar.exampl.yml]: Done.

File: cribl-edge/config/log.sls (hunk 4 above)
Comment (ferricoxide) [reply cribl-edge/config/log.sls]: Fixed.

File: cribl-edge/package/clean.sls (hunk 6 above)
Comment (ferricoxide) [reply cribl-edge/package/clean.sls]: Fixed

**Review by lorengordon — APPROVED** [review]

**Pull request merged**